design-documents/graph-ql/coverage/company.md

```diff
@@ -0,0 +1,527 @@
+    id: Int @doc(description: "Company Administrator's ID.")
+    email: String @doc(description: "Company Administrator email address.")
+    firstname: String @doc(description: "Company Administrator first name.")
+    lastname: String @doc(description: "Company Administrator last name.")
+    job_title: String @doc(description: "Company Administrator job title.")
+    gender: Int @doc(description: "Company Administrator gender.")
+}
+
+type CompanyAddressRegion @doc(description: "Output data schema for a region data in Company address.") {
+    region_id: Int @doc(description: "Unique region identifier. Required for certain countries.")
+    region: String @doc(description: "State or province name.")
+    region_code: String @doc(description: "Region code.")
+}
+
+type CompanySalesRepresentative @doc(description: "Company sales representative information output data schema.") {
+    email: String @doc(description: "Sales representative email address.")
+    firstname: String @doc(description: "Sales representative first name.")
+    lastname: String @doc(description: "Sales representative last name.")
+}
+
+type CompanyUsers @doc(description: "Output data schema for an object returned by a Company users search query.") {
+    items: [CompanyUser] @doc(description: "An array of 'CompanyUser' objects that match the specified search criteria.")
+    total_count: Int @doc(description: "The number of objects returned.")
+    page_info: SearchResultPageInfo @doc(description: "An object that includes the pagination meta data of the search query.")
+}
+
+type CompanyUser @doc(description: "Company User (Customer assigned to Company) entity output data schema.") {
+    id: Int @doc(description: "Company User/Customer ID.")
+    email: String @doc(description: "Company User email address.")
+    firstname: String @doc(description: "Company User first name.")
+    lastname: String @doc(description: "Company User last name.")
+    status: Int @doc(description: "Company User status ID.")
+    status_name: String @doc(description: "Company User status name.")
+    job_title: String @doc(description: "Company User job title.")
+    telephone: String @doc(description: "Company User phone number.")
+    role: CompanyRole @doc(description: "Company User role data (includes permissions).")
+    team: CompanyTeam @doc(description: "Company User team data.")
+}
+
+type CompanyRoles @doc(description: "Output data schema for an object returned by a Company roles search query.") {
+    items: [CompanyRole] @doc(description: "An array of 'CompanyUser' objects that match the specified search criteria.")
+    total_count: Int @doc(description: "The number of objects returned.")
+    page_info: SearchResultPageInfo @doc(description: "An object that includes the pagination meta data of the search query.")
+}
+
+type CompanyRole @doc(description: "Company role output data schema returned in response to a query by Role ID.") {
+    id: Int @doc(description: "Role ID.")
```

Comment:
Please search for all ID fields in the schema and change the type to `ID`. We also need to explain in this proposal, how this ID should be generated. In other new schema's we say it should be `base64` from some other existing ID. And in the future the hope is to change current value to the UUID 
```suggestion
    id: ID @doc(description: "Role ID.")
```

Reply:
fixed